pkg/server/server.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"syscall"
+
+	"github.com/go-logr/logr"
+	"github.com/oklog/run"
+
+	"github.com/NVIDIA/knavigator/pkg/config"
+	"github.com/NVIDIA/knavigator/pkg/engine"
+)
+
+type Server struct {
+	s   *http.Server
+	log *logr.Logger
+}
+
+type TaskHandler struct {
```

Comment:
Considering the API is simple, instead of using structs, can we use functions as HTTP handlers? Please https://www.alexedwards.net/blog/an-introduction-to-handlers-and-servemuxes-in-go (section "Functions as handlers")